proxy/proxy.go

```diff
@@ -226,31 +234,48 @@ func (p *Proxy) fuse(client, backend net.Conn) {
 	defer p.logConnectionMessage("closed", client, backend)
 	p.logConnectionMessage("opening", client, backend)
 
-	go func() { p.copyData(client, backend) }()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() { p.copyData(client, backend); wg.Done() }()
 	p.copyData(backend, client)
+	wg.Wait()
```

Comment:
It's cleaner this way -- otherwise the "closed pipe" log line might be printed before both ends are actually closed. Fine in general because if one closes the other one does, so it's not like we're leaking goroutines, but I like it better this way.

Reply:
makes sense, thanks!